src/applications/gqrx/mainwindow.cpp

```diff
@@ -1586,6 +1584,8 @@ void MainWindow::setIqFftSize(int size)
 {
     qDebug() << "Changing baseband FFT size to" << size;
     rx->set_iq_fft_size(size);
+    for (int i = 0; i < size; i++)
+        d_iirFftData[i] = 0;
```

Comment:
I noticed that the averaged FFT data was not cleared when changing FFT sizes, which results in spurious data being displayed until the old data fades away. This fixes the problem by zeroing the averages when the FFT size is changed.